pallets/dedir/src/types.rs

```diff
@@ -0,0 +1,99 @@
+// This file is part of CORD – https://cord.network
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// Copyright (C) Dhiway Networks Pvt. Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Adapted to meet the requirements of the CORD project.
+
+// CORD is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// CORD is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with CORD. If not, see <https://www.gnu.org/licenses/>.
+
+//! DeDir pallet types.
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+
+use sp_runtime::RuntimeDebug;
+use sp_std::prelude::*;
+
+#[derive(Encode, Decode, MaxEncodedLen, Clone, RuntimeDebug, PartialEq, Eq, TypeInfo)]
+pub enum RegistrySupportedStateOf {
```

Comment:
Do we need a DRAFT state at the chain level? Without that this can be a bool filed representing the status

Reply:
This thought process came when we asked the question if chain alone is used as 'store' also, how to differentiate between something which is not ready to be 'lookedup' (or queried), and the one record which is ready to do the lookup.

Having these states covers most of the usecases is what we thought.